xla/service/scatter_determinism_expander.cc

```diff
@@ -368,15 +376,6 @@ static absl::StatusOr<HloInstruction*> CreateScanWithIndices(
     auto* indices_mask = parent->AddInstruction(HloInstruction::CreateCompare(
         ShapeUtil::MakeShape(PRED, {num_updates}), indices,
         concatenated_indices, ComparisonDirection::kEq));
-    auto* first_element_false = parent->AddInstruction(
```

Comment:
Makes sense to remove this now. When I saw this fix, I was a bit sceptical it would work in all cases, which is how I came up with the additional test. It was a good quick fix though for the "normal" cases.